sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkAsyncHttpClientBuilder.java

```diff
@@ -144,7 +182,56 @@ public HttpClient build() {
                         buildProxyOptions.getPassword()));
             }
         }
-        return new JdkAsyncHttpClient(httpClientBuilder.build());
+        return new JdkAsyncHttpClient(httpClientBuilder.build(), Collections.unmodifiableSet(getRestrictedHeaders()));
+    }
+
+    Set<String> getRestrictedHeaders() {
+        // Compute the effective restricted headers by removing the allowed headers from default restricted headers
+        Set<String> allowRestrictedHeaders = getAllowRestrictedHeaders();
+        Set<String> restrictedHeaders = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+        restrictedHeaders.addAll(DEFAULT_RESTRICTED_HEADERS);
+        restrictedHeaders.removeAll(allowRestrictedHeaders);
+        return restrictedHeaders;
+    }
+
+    private Set<String> getAllowRestrictedHeaders() {
+        Properties properties = getNetworkProperties();
+        String[] allowRestrictedHeadersNetProperties =
+            properties.getProperty(JDK_HTTPCLIENT_ALLOW_RESTRICTED_HEADERS, "").split(",");
+
+        // Read all allowed restricted headers from configuration
+        Configuration config = (this.configuration == null)
```

Comment:
Does the JVM check both environment and system properties when allowing these headers to be used? If not we should use whichever it uses directly as `Configuration` will lookup against both.

Reply:
Since most of our proxy and other configurations are checking both, users might want to use the same location (either env var or sys properties) for all their configurations. So, checking both as it makes it easier for the user.